### PR TITLE
Support ImagePullSecrets in Executor pod

### PR DESCRIPTION
**Pull request by k0da**
Signed-off-by: Dinar Valeev <dinar.valeev@absa.africa>

### CODE REVIEW

**Review by somaritane — CHANGES_REQUESTED** [review]
@k0da, LGTM, I'd probably add a few words about this feature into README.md.

File: pkg/terraform/state/deploy.go [hunk 1]
```diff
@@ -258,6 +259,9 @@ func (h *handler) createJob(or []metaV1.OwnerReference, input *Input, runName, r
 			Template: coreV1.PodTemplateSpec{
 				Spec: coreV1.PodSpec{
 					ServiceAccountName: sa,
+					ImagePullSecrets: []coreV1.LocalObjectReference{
```
Comment (somaritane) [inline pkg/terraform/state/deploy.go:262]: If you put ImagePullSecrets into service account used for pod creation, you don't have to provide it in the pod spec:
https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#add-image-pull-secret-to-service-account

Comment (k0da) [reply pkg/terraform/state/deploy.go:262]: I don't see the difference, we create both SA and Job, both are unique per Execution.

**Review by somaritane — COMMENTED** [review]
@k0da LGTM!

**Review by somaritane — APPROVED** [review]

**Pull request merged**